test/parallel/test-stream-readable-unshift.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const { Readable } = require('stream');
+
+{
+  // Check that strings are saved as Buffer
+  const readable = new Readable({ read() {} });
+
+  const string = 'abc';
+  readable.unshift(string);
+
+  assert.strictEqual(
+    Buffer.isBuffer(readable._readableState.buffer.head.data),
+    true
+  );
+
+  // Check default encoding is utf8
+  assert.strictEqual(
+    readable._readableState.buffer.head.data.toString('utf8'),
```

Comment:
This accesses a lot of internals. Is it also possible to test for these things with public APIs only?

Reply:
Unfortunately, the only way to check if it was saved correctly in the `BufferList` is accessing the state, which is internal. Where it was possible to test using public APIs, I used those instead:

https://github.com/nodejs/node/blob/22e814bd2cf0c73c4850ea08fde10f6792930750/test/parallel/test-stream-readable-unshift.js#L97-L102

All the tests I've seen for streams that need to check the behaviour of the state, they check it accessing the internal properties.

Reply:
I’m not in agreement with that argument. It’s possible to verify that the data is coming out of the stream with the correct encoding. After all, if this change was not visible from a public API, you would not be making it.

Reply:
You're right, I thought some things couldn't be tested without the use of internals. I will submit the changes soon.